<?xml version="1.0" ?>

<launch>
    <arg name="robot_type" default="$(env ROBOT_TYPE)" doc="Robot type: [lt, go1]"/>
    <arg name="rl_config_file" default="$(eval 'go1_student.yaml' if robot_type == 'go1' else 'lt_student.yaml')" />


    <include file="$(find rl_controllers)/launch/joy_teleop.launch"/>

    <rosparam file="$(find rl_controllers)/config/$(env ROBOT_TYPE)_kd.yaml" command="load"/>

    <param name="policyModelPath" value="$(find rl_controllers)/policy/$(env ROBOT_TYPE)/policy_lstm.onnx"/>
    <param name="encoderModelPath" value="$(find rl_controllers)/policy/$(env ROBOT_TYPE)/lstm_encoder.onnx"/>



    <rosparam file="$(find rl_controllers)/config/controllers.yaml" command="load"/>
    <rosparam file="$(find rl_controllers)/config/$(arg rl_config_file)" command="load"/>

    <node name="controller_spawner" pkg="controller_manager" type="spawner" 
        respawn="false" output="screen" args="
            controllers/student_policy_controller"/>

            

    <node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure" output="screen">
    </node>
</launch>
